### DISCUSSION TITLE
Preferred way to keep personal configs on git with minimal-emacs.d?

### DISCUSSION THREAD

**Discussion by jhiltu** [jamescherti/minimal-emacs.d#13] [Q&A] [answered]
Hey, thanks for the great config! This is exactly what I was looking for to start building my own Emacs config. No need to write tons of config just to get the editor barely usable but not too much going on either.

I'm curious what is the preferred way to keep my own configs (`post-init.el`, `pre-early-init.el` etc) on git also but also keep getting potential updates from this repo?

Now I did it with symbolic links and making a new separate git repository that only holds my personal configs. Is this okay or can I run into issues down the line? Better to fork and remove the ignored files from .gitignore? Or both work ok? Newish to version control and git!

Could the absolutely excellent README also have small section about this topic?

(below is what I'm doing right now)
```
git clone https://github.com/jamescherti/minimal-emacs.d ~/.config/emacs
ln -s ~/.config/min-emacs/pre-early-init.el ~/.config/emacs/pre-early-init.el
ln -s ~/.config/min-emacs/post-init.el ~/.config/emacs/post-init.el
```

**Comment by jamescherti** [comment, marked as answer]
Hello @jhiltu,

I am glad to hear the `minimal-emacs.d` config has been helpful to you!

The symbolic link solution you're using is actually a good approach. It keeps the configuration separate while still allowing you to easily pull updates from the original repository. 

Alternative solution: Modify the `minimal-emacs-user-directory` variable by adding the following to the `~/.emacs.d/pre-early-init.el` file:
```elisp
(setq minimal-emacs-user-directory "~/.config/min-emacs/")
```
This will ensure that the `minimal-emacs.d` configuration loads `post-early-init.el`, `pre-init.el`, and `post-init.el` from `~/.config/min-emacs/`.
